mmdet/core/evaluation/eval_hooks.py

```diff
@@ -10,17 +10,31 @@ class EvalHook(Hook):
     Attributes:
         dataloader (DataLoader): A PyTorch dataloader.
         interval (int): Evaluation interval (by epochs). Default: 1.
+        start (int, optional): Evaluation starting epoch. It enables evaluation
+            before the training starts if ``start`` <= the resuming epoch.
+            If None, whether to evaluate is merely decided by ``interval``.
+            Default: None.
+        **eval_kwargs: Evaluation arguments fed into the evaluate function of
+            the dataset.
     """
 
-    def __init__(self, dataloader, interval=1, **eval_kwargs):
+    def __init__(self, dataloader, interval=1, start=None, **eval_kwargs):
         if not isinstance(dataloader, DataLoader):
             raise TypeError('dataloader must be a pytorch DataLoader, but got'
                             f' {type(dataloader)}')
         self.dataloader = dataloader
         self.interval = interval
+        self.start = start
         self.eval_kwargs = eval_kwargs
 
+    def before_run(self, runner):
```

Comment:
Better to use `before_train_epoch` instead of `before_run`. There may be custom hooks that is required before running the evaluation.

Reply:
I thought about that too. But adding it to `before_train_epoch` would cause two repeated evaluations. Besides, logger hook level is "VERY_LOW" and chances are rare that one would put a custom hook about dataset handling after this one.